File: backend/src/main/java/com/dmart/service/CartService.java
package com.dmart.service;

import com.dmart.dto.request.AddToCartRequestDto;
import com.dmart.dto.request.UpdateCartItemRequestDto;
import com.dmart.dto.response.CartResponseDto;

public interface CartService {

    CartResponseDto getMyCart();

    CartResponseDto addToCart(AddToCartRequestDto dto);

    CartResponseDto updateCartItem(Long cartItemId, UpdateCartItemRequestDto dto);

    CartResponseDto removeCartItem(Long cartItemId);

    CartResponseDto clearCart();
}

File: frontend/src/services/cartService.js
import { request } from './api';

export const cartService = {
  getCart: () => request('/cart'),
  addToCart: (productId, quantity = 1) => request('/cart/items', { method: 'POST', body: JSON.stringify({ productId, quantity }) }),
  updateCartItem: (itemId, quantity) => request(`/cart/items/${itemId}`, { method: 'PUT', body: JSON.stringify({ quantity }) }),
  removeCartItem: (itemId) => request(`/cart/items/${itemId}`, { method: 'DELETE' }),
  clearCart: () => request('/cart', { method: 'DELETE' }),
};